pkg/debuginfo/extract.go

```diff
@@ -79,6 +79,13 @@ func (e *Extractor) Extract(ctx context.Context, dst io.WriteSeeker, src string)
 	if err != nil {
 		return fmt.Errorf("failed to initialize writer: %w", err)
 	}
+
+	// if ELF file doesn't contain `.text` section then there is no need to strip the file.
+	if !w.Writer.HasSection(".text") {
```

Comment:
🔧 I think we don't need to add this method to the `Writer` interface. We can just check the raw ELF file to see whether it has the necessary sections and keep it as is. Since we already have an open file, we can just use `https://pkg.go.dev/debug/elf#NewFile` and quickly check the existence of the `.text` section and early return.

Also, I think doing this check even before the `Extract` call is better.